SETUP/smoketests/pageload-smoketest.sh

```diff
@@ -0,0 +1,253 @@
+    'quiz/tuts/tut_p_aeoe_2.php'
+    'quiz/tuts/tut_p_basic_1.php'
+    'quiz/tuts/tut_p_basic_2.php'
+    'quiz/tuts/tut_p_basic_3.php'
+    'quiz/tuts/tut_p_basic_4.php'
+    'quiz/tuts/tut_p_basic_5.php'
+    'quiz/tuts/tut_p_fraktur.php'
+    'quiz/tuts/tut_p_greek_1.php'
+    'quiz/tuts/tut_p_greek_2.php'
+    'quiz/tuts/tut_p_greek_3.php'
+    'quiz/tuts/tut_p_greek_4.php'
+    'quiz/tuts/tut_p_greek_5.php'
+    'quiz/tuts/tut_p_mod1_1.php'
+    'quiz/tuts/tut_p_mod1_2.php'
+    'quiz/tuts/tut_p_mod1_3.php'
+    'quiz/tuts/tut_p_mod1_4.php'
+    'quiz/tuts/tut_p_mod1_5.php'
+    'quiz/tuts/tut_p_mod2_1.php'
+    'quiz/tuts/tut_p_mod2_2.php'
+    'quiz/tuts/tut_p_mod2_3.php'
+    'quiz/tuts/tut_p_mod2_4.php'
+    'quiz/tuts/tut_p_mod2_5.php'
+    'quiz/tuts/tut_p_old_1.php'
+    'quiz/tuts/tut_p_old_2.php'
+    'quiz/tuts/tut_p_old_3.php'
+    'quiz/tuts/tut_p_thorn.php'
+)
+
+TEAMS_URLS=(
+    'stats/members/jointeam.php?tid=44' # NB performs an action
+    'stats/members/quitteam.php?tid=44' # NB performs an action
+    'stats/teams/team_topic.php?team=44' # Adds a forum post!
+    'stats/members/mbr_list.php?tid=44'
+    'stats/members/mbr_xml.php?username=teststeel'
+    'stats/members/mdetail.php?id=1'
+    'stats/teams/new_team.php'
+    'stats/teams/tdetail.php?tid=44'
+    'stats/teams/teams_xml.php?tid=44'
+    'stats/teams/tedit.php?tid=44'
+    'stats/teams/tlist.php'
+)
+STATS_URLS=(
+    'stats/PP_unknown.php'
+    'stats/default.php'
+    'stats/equilibria.php'
+    'stats/misc_stats1.php?tally_name=F1&start=2024-01&end=2024-02'
+    'stats/misc_user_graphs.php'
+    'stats/pages_in_states.php'
+    'stats/pages_proofed_graphs.php?tally_name=P3' # Slow to evaluate!
+    'stats/percent_users_who_proof.php'
+    'stats/pm_stats.php'
+    'stats/pp_stage_goal.php'
+    'stats/pp_stats.php'
+    'stats/ppv_stats.php'
+    'stats/projects_Xed_graphs.php?which=posted'
+    'stats/proof_stats.php?tally_name=F2'
+    'stats/release_queue.php'
+    'stats/requested_books.php'
+    'stats/round_backlog.php'
+    'stats/round_backlog_days.php'
+    'stats/stats_central.php'
+    'stats/user_logon_graphs.php'
+)
+STYLES_URLS=(
+    'styles/design_philosophy.php'
+    'styles/style_demo.php'
+)
+BASE_TOOL_URLS=(
+    'tools/change_sr_commitment.php?projectid=projectID5e23a810ef693&action=commit'
+)
+
+URLS=(
+    ${NOLOGIN_URLS[*]}
+    ${BASE_URLS[*]}
+    ${API_URLS[*]}
+    ${FAQ_URLS[*]}
+    ${MISC_URLS[*]}
+    ${QUIZ_URLS[*]}
+    ${TEAMS_URLS[*]}
+    ${STATS_URLS[*]}
+    ${STYLES_URLS[*]}
+    ${BASE_TOOL_URLS[*]}
+)
+
+# TODO Add remaining ~95 pages
+# TODO Add accounts/logout.php
+
+function do_exit()
+{
+    if [ $ERR != 0 ]; then
+        echo Exiting with error
+        echo Server log:
+        cat $SERVER_LOG
+    fi
+    # Kill all children of this script (ie the php -S, if it's still alive)
+    /bin/kill -INT $(jobs -p)
+}
+
+trap do_exit EXIT
+
+# HACK: Give php -S time to start up
+sleep .5
```

Comment:
no way to do this with polling perhaps or some other mechanism so this step isn't flakey on a slow run?

Reply:
Heh. Good idea. Nothing stopping us injecting a `helloworld.php` into the root directory and polling that for the purposes of this test.

Reply:
I ended up rewriting this completely in Python because it was getting to complex to manage as a shell script. It now polls at startup that the webserver is running.